tst/testlibgap/0001.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Small program to test libgap linkability and basic working
+ */
+#include <stdio.h>
+#include <unistd.h>
+
+#include <compiled.h>
+#include <libgap-api.h>
+
+extern char ** environ;
+
+void test_eval(const char * cmd)
+{
+    Obj res, ires;
+    Int rc, i;
+
+    printf("Input:\n%s", cmd);
+    res = GAP_EvalString(cmd);
+    rc = GAP_LenPList(res);
+    for(i=1;i<=rc;i++) {
+        ires = GAP_ElmPList(res, i);
+        if( GAP_ElmPList(ires, 1) == True )
+            printf("Output:\n%s\n", CSTR_STRING(GAP_ElmPList(ires, 5)));
+    }
+}
+
+int main(int argc, char **argv)
+{
+    printf("Initializing GAP...\n");
+    GAP_Initialize(argc, argv, environ, 0L, 0L);
+    printf("done\n");
+
+    CollectBags(0, 1);    // full GC
+
+    test_eval("1+2+3;\n");
+    test_eval("g:=FreeGroup(2);\n");
+    test_eval("a:=g.1;\n");
+    test_eval("b:=g.2;\n");
+    test_eval("lis:=[a^2, a^2, b*a];\n");
+    test_eval("h:=g/lis;\n");
+    test_eval("c:=h.1;\n");
+    test_eval("Set([1..1000000], i->Order(c));\n");
+
+    GAP_Finalize();
+    return 0;
+}
+
+
+/*
+g:=FreeGroup(2);
+a:=g.1;
+b:=g.2;
+lis:=[a^2, a^2, b*a];
+h:=g/lis;
+c:=h.1;
+Set([1..300000], i->Order(c));
+ */
```

Comment:
Is this supposed to be the same code as in the test above, which then is supposed to be comparable via copy&paste into a GAP session?

Then I think this is a bad idea, as the two will quickly run out of sync. Of course it's not my problem for now, so I'll not overly complain, but why not do this instead: Put this code into a separate `.g` file, then use fopen + fgets + fclose (or open + gets + close) to process that file line bby line from within this .c file.
Or, if you prefer to avoid dealing with opening files, just read from stdin line-by-line, and change the test script to pipe the `.g` file into the test executable.

Reply:
In principle the test only needs to execute one string, wherever it comes from. If we can get the kernel tests to work (#1917).

This is not a particularly good test, but the initial testing example that I either copied from SageMath or wrote in France 18 months ago...

Real tests would surely want to make sure that the (to be defined) API works as expected.